src/config/models.rs

```diff
@@ -1,6 +1,188 @@
-/// Defines which gpt model to use. Currently set to "gpt-4o"
-pub const MODEL: &str = "gpt-4o";
-/// Defines default maximum number of tokens available in conversation and response
-pub const MAX_TOKENS: &i32 = &4096;
-/// Defines default temperature of response
-pub const TEMPERATURE: &f32 = &0.6;
+use memoize::memoize;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::PathBuf;
+
+fn default_api_url() -> String {
+    "https://api.openai.com/v1/".to_string()
+}
+fn default_model() -> String {
+    "gpt-4o".to_string()
+}
+fn default_max_tokens() -> i32 {
+    4096
```

Comment:
```suggestion
    8192
```
Unrelated to your PR, but required to get CI to pass with this size PR!